[Rajarshi Basu] iP

rajobasu opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

rajobasu committed:
Added Basic structure of IOManager, can only echo input except "bye".

rajobasu committed:
Added some formatting.

rajobasu committed:
Added support for addition and listing

Added support for more functionalities. The matching of operation type is currently handled by the InputType enum, while the action is handled by the InputHandler. TODO : combine the action handler in with the enum itself.

rajobasu committed:
Added support for Done function

The enums now handle both checking as well as doing the action they are associated with in the form of lambdas. InputHandler just passes in those functions to the enums.

rajobasu committed:
Added todos, events and deadlines.

rajobasu committed:
Added semi-automatic testing

rajobasu committed:
Exception Handling for Done, TODO, and UNKNOWN added.

rajobasu committed:
Deletion support added. Bug fixed regarding task done.

rajobasu committed:
Add support for Persistent Storage

Previously everything was stored in the temporary memory.
Created an interface for saveing the task list, and made an implementation where everything is stored in a local file.

rajobasu committed:
Add support for LocalDate in Deadline

Previously, entered date was not interpreted as a date.

Now added support and made it an actual LocalDate object.

rajobasu committed:
Merge branch 'branch-Level-7'

rajobasu committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	.gitignore

rajobasu committed:
Organize into more packages

Earlier all the code was in the core package.
Created another package that specifically contain task related classes and interfaces.

rajobasu committed:
Add Testing support.

Added some simple testing for a few public methods in TaskManager, and TodoTask creation.

Also updated TaskManager to have a boolean parameter based on which it will load from storage or start a new file altogether.

rajobasu committed:
Add documentation, update .gitignore

Added documentation to the public methods. Updated gitignore to ignore the manifest files from JAR creation.

rajobasu committed:
Change to follow Java Conventions

rajobasu committed:
Add support for find

Added support for keyword finding, and fixed bug w.r.t loading fails
when storage file throws ClassCastException.

rajobasu committed:
Add support for find

rajobasu committed:
Change method name to match Coding Standard

Change the name of getDone to setDone in Task.java

rajobasu committed:
Change class to match Java Coding Standard

rajobasu committed:
Add Java Doc

rajobasu committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into A-Gradle

rajobasu committed:
Add Gradle support

rajobasu committed:
Add bugfix for gradle file

rajobasu committed:
Use Var-args in main syntax

rajobasu committed:
Add dependency for javafx

rajobasu committed:
Add GUI using javafx

rajobasu committed:
Add support for JavaFX and GUI

chanellNg commented on src/main/java/core/IOManager.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * Creates a new IOManager using specified input and output streams.
```
Perhaps an empty line between description and parameter section could be added.

prerthan99 commented on src/main/java/core/IOManager.java:
```diff
@@ -1,22 +1,37 @@
         this.is = is;
         this.pw = new PrintWriter(ps);
     }
 
```
It would be a good idea to explain what the param id for i.e. explain what listener is for and what it does

prerthan99 commented on src/main/java/core/InputListener.java:
```diff
@@ -1,5 +1,14 @@
+/**
+ * Functional Interface which only has a function to deal with new method.
+ */
 public interface InputListener {
```
the input names are obvious so good job there, but the explanation doesnt really add anything to the javadocs

prerthan99 commented on src/main/java/core/TaskStorageFile.java:
```diff
@@ -11,7 +11,10 @@
 import java.nio.file.Path;
```
I usually comment in the beginning what functions each import does

chanellNg commented on src/main/java/core/InputListener.java:
```diff
@@ -0,0 +1,14 @@
+public interface InputListener {
+    /**
+     * Method to deal with any input.
+     * @param type - type of input
```
Maybe could add punctuations to the Javadoc comments

prerthan99 commented on src/main/java/core/task/Task.java:
```diff
@@ -16,35 +19,64 @@
+     * Returns whether the task has already been done or not.
+     * @return - whether the task has been done.
+     */
     public boolean isDone() {
```
I read somewhere that its better to name these getter methods getterIsDone

chanellNg commented on src/main/java/core/TaskStorageFile.java:
```diff
@@ -0,0 +1,47 @@
+    @Override
+    public ArrayList<Task> retrieveTaskList(Path fileLocation) throws IOException {
+        if (Files.isRegularFile(fileLocation)) {
+            try (var fWriter = new FileInputStream(fileLocation.toFile()); var ois = new ObjectInputStream((fWriter))) {
```
Perhaps a line break could be added here.

chanellNg commented on src/main/java/core/task/TaskManager.java:
```diff
@@ -0,0 +1,169 @@
+    public void reloadTaskList() {
+        try {
+            allTasks = taskStorage.retrieveTaskList(savePath);
+
```
Perhaps could remove the line break for the try-catch statement.

chanellNg reviewed (commented):
Overall, the naming was very consistent and the Javadoc comments were very informative and helpful. Indentation was also consistent.  I liked the use of an enumeration for the input types.

rajobasu committed:
Add asserts

rajobasu committed:
Fix suppressWarnings in TaskStorage

rajobasu committed:
Merge pull request #1 from rajobasu/branch-A-CodeQuality

Fix suppressWarnings in TaskStorage

rajobasu committed:
Accept new master changes

rajobasu committed:
Merge pull request #2 from rajobasu/branch-A-Assertions

Add asserts

rajobasu committed:
Add support for partial word search

rajobasu committed:
Merge branch 'branch-C-BetterSearch'

rajobasu committed:
Minor edits

Edits like changing some variables to final, etc. Doesn't effectively change anything.

rajobasu committed:
Update TodoTest import statements

rajobasu committed:
Change Gradle to fix shadowJar archiveBaseName

rajobasu committed:
Add credit notes

rajobasu committed:
Update LocalDate handling

rajobasu committed:
Add and update documentation

rajobasu committed:
Abstract DateTime related features to common superclass

rajobasu committed:
Add UI.png

rajobasu committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/ui/DukeApplicationPage.java

rajobasu committed:
Rename project showcase screenshot

rajobasu committed:
Add User Guide

rajobasu committed:
Update User Guide
